hack/license_boilerplate.txt

```diff
@@ -0,0 +1,3 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
```

Comment:
Missing `year`.

While @unmarshall and I were working on [etcd-druid#728](https://github.com/gardener/etcd-druid/issues/728), and were rebasing on latest etcd-druid master, we saw that the year in the boilerplate.txt file was hardcoded by [this PR](https://github.com/gardener/etcd-druid/pull/770) from you. We made a couple of changes to make it work for any given year, in [this commit](https://github.com/unmarshall/etcd-druid/commit/3c550a3a3ee0afa775b087bab39426be2c412044). Please incorporate something similar for this PR as well.

Essentially, you need to first remove the `/`s in the boilerplate file, and use sed to replace the `{YEAR}` with the actual year, like we've done [here](https://github.com/unmarshall/etcd-druid/commit/3c550a3a3ee0afa775b087bab39426be2c412044#diff-bc5efa1d5f2c010c2f712560dead0417d82695f7158075a80ba276abf6003d4c). Then, the Makefile target needs to be enhanced to pass a `YEAR` parameter to the script, like [this](https://github.com/gardener/etcd-druid/blob/9c5f8254e3aeb24c1e3e88d17d8d1de336ce981b/Makefile#L145-L147), and finally run the make target as `YEAR=2024 make add-license-headers`.

```suggestion
// SPDX-FileCopyrightText: {YEAR} SAP SE or an SAP affiliate company and Gardener contributors
```

Reply:
regarding https://github.com/gardener/etcd-druid/pull/770
My intend was to not include the year at all in hack/boilerplate.go.txt ! That's actually a mistake on my side